mne/decoding/receptive_field.py

```diff
@@ -201,18 +203,33 @@ def fit(self, X, y):
 
         # zero-pad if necessary
         if isinstance(self.estimator, TimeDelayingRidge):
-            X = _pad_time_series(X, n_delays=len(self.delays_),
+            X = _pad_time_series(X, n_delays=n_delays,
                                  fill_mean=self.fit_intercept)
-            y = _pad_time_series(y, n_delays=len(self.delays_),
+            y = _pad_time_series(y, n_delays=n_delays,
                                  fill_mean=self.fit_intercept)
+
         # convert to sklearn and back
         X, y = self._delay_and_reshape(X, y)
         self.estimator_.fit(X, y)
-        del X, y
         coef = get_coef(self.estimator_, 'coef_')  # (n_targets, n_features)
-        shape = [n_feats, len(self.delays_)]
+        shape = [n_feats, n_delays]
         shape = ([-1] if self._y_dim > 1 else []) + shape
         self.coef_ = coef.reshape(shape)
+
+        # Inverse-transform model weights.
+        # inv_coef has shape = (n_feats * n_delays, n_outputs)
+        coef = np.reshape(self.coef_, (n_feats * n_delays, n_outputs))
+        if not isinstance(self.estimator_, TimeDelayingRidge):
+            inv_coef = np.linalg.multi_dot([np.dot(X.T, X), coef,
+                                            np.linalg.inv(np.dot(y.T, y))])
+        else:
+            inv_coef = np.linalg.multi_dot([self.estimator_.x_xt_, coef,
+                                            np.linalg.inv(np.dot(y.T, y))])
```

Comment:
You need to ensure that y is centered to 0. 

Prefer using np.cov when possible to simplify equation